### ISSUE TITLE
Context manager confuses branch miss detector when de-dent is close to finally statement

### ISSUE THREAD

**Issue by nedbat** [coveragepy/coveragepy#473]
*Originally reported by* **bossylobster (Bitbucket: [bossylobster](https://bitbucket.org/bossylobster), GitHub: Unknown)**

----------------------------------------

Running the code to [reproduce][1] gives:

```python
$ coverage run --branch repro.py
{}
$ coverage report -m
Name       Stmts   Miss Branch BrPart  Cover   Missing
------------------------------------------------------
repro.py      16      0      2      1    94%   17->20
```

![screen_shot_001.png](https://bitbucket.org/repo/Ee7X/images/3783599554-screen_shot_001.png)

----

Maybe I am misunderstanding which branches should occur?

[1]: https://gist.github.com/dhermes/823dd28cfa37289769f3

----------------------------------------
- Bitbucket: https://bitbucket.org/ned/coveragepy/issue/473

**Comment by nedbat** [comment]
Duplicate of #146.

**Comment by nedbat** [comment]
This is fixed in 4.1beta.

**Comment by nedbat** [comment]
*Original comment by* **bossylobster (Bitbucket: [bossylobster](https://bitbucket.org/bossylobster), GitHub: Unknown)**

----------------------------------------

My bad. I only looked in open issues for dupes. Thanks for the quick response!
